src/BinSkim.Driver/AnalyzeOptions.cs

```diff
@@ -43,5 +43,15 @@ public class AnalyzeOptions : AnalyzeOptionsBase
             HelpText = "Emit verbose output. The resulting comprehensive report is designed to provide appropriate evidence for compliance scenarios.")]
         [Obsolete("Use --level and --kind instead.")]
         public bool Verbose { get; set; }
+
+        [Option(
+            "repository-uri",
+            HelpText = "RepositoryUri is required to rule BA4001 and BA4002.")]
```

Comment:
>RepositoryUri is required to rule BA4001 and BA4002 [](http://example.com/codeflow?start=24&length=51)

Whenever you show an opaque rule id to the user, you should add the readable name. Not everyone can remember what BA4001 is. Also, you've described a fact here 'this data is required to run rule BA4001.ReportPECompilerData' but you haven't described what that data is.

What is this data? Does it make sense that we require a single repo URL for a scan run??

Reply:
I changed the helptext. Let me know what do you think.

About your question...I thought about it, and  I could not find anything better based on this:
[https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml](https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml)

my idea was  to find one pre-defined variable that any pipeline would be  able to get.